### PR TITLE
[PWGDQ] Fixes in dqEfficiency_withassoc and adding muon refit in tableMakerMC

### PR DESCRIPTION
**Pull request by mcoquet642**
- Adding checks in dqEfficiency to make sure considered particle has an associated mcparticle
- Adding refit option for global muons in tableMakerMC
- removing protection on global muon covariance in FillPairVertexing, this is fixed when using the option "UseAbsDCA" in the configuration of the DCAFitter

### CODE REVIEW

File: PWGDQ/TableProducer/tableMakerMC_withAssoc.cxx [hunk 1]
```diff
@@ -159,11 +159,14 @@ struct TableMakerMC {
 
   // CCDB connection configurables
   Configurable<std::string> fConfigCcdbUrl{"ccdb-url", "http://alice-ccdb.cern.ch", "url of the ccdb repository"};
-  Configurable<bool> fPropMuon{"cfgPropMuon", false, "Propgate muon tracks through absorber"};
   Configurable<std::string> geoPath{"geoPath", "GLO/Config/GeometryAligned", "Path of the geometry file"};
   Configurable<std::string> grpmagPath{"grpmagPath", "GLO/Config/GRPMagField", "CCDB path of the GRPMagField object"};
   Configurable<std::string> grpmagPathRun2{"grpmagPathRun2", "GLO/GRP/GRP", "CCDB path of the GRPObject (Usage for Run 2)"};
 
+  // Muon related options
+  Configurable<bool> fPropMuon{"cfgPropMuon", true, "Propgate muon tracks through absorber (do not use if applying pairing)"};
```
Comment (singiamtel) [inline PWGDQ/TableProducer/tableMakerMC_withAssoc.cxx:167]: Small typo on the copy paste

```suggestion
  Configurable<bool> fPropMuon{"cfgPropMuon", true, "Propagate muon tracks through absorber (do not use if applying pairing)"};
```

File: PWGDQ/TableProducer/tableMaker_withAssoc.cxx [hunk 2]
```diff
@@ -212,6 +212,7 @@ struct TableMaker {
 
   // Muon related options
   Configurable<bool> fPropMuon{"cfgPropMuon", true, "Propgate muon tracks through absorber (do not use if applying pairing)"};
```
Comment (singiamtel) [inline PWGDQ/TableProducer/tableMaker_withAssoc.cxx:214]:
```suggestion
  Configurable<bool> fPropMuon{"cfgPropMuon", true, "Propagate muon tracks through absorber (do not use if applying pairing)"};
```

File: PWGDQ/TableProducer/tableMaker_withAssoc.cxx [hunk 3]
```diff
@@ -212,6 +212,7 @@ struct TableMaker {
...
+  Configurable<bool> fRefitGlobalMuon{"cfgRefitGlobalMuon", true, "Propgate muon tracks through absorber (do not use if applying pairing)"};
```
Comment (singiamtel) [inline PWGDQ/TableProducer/tableMaker_withAssoc.cxx:215]:
```suggestion
  Configurable<bool> fRefitGlobalMuon{"cfgRefitGlobalMuon", true, "Propagate muon tracks through absorber (do not use if applying pairing)"};
```

File: PWGDQ/TableProducer/tableMakerMC_withAssoc.cxx [hunk 4]
```diff
@@ -159,11 +159,14 @@ struct TableMakerMC {
...
+  Configurable<bool> fRefitGlobalMuon{"cfgRefitGlobalMuon", true, "Propgate muon tracks through absorber (do not use if applying pairing)"};
```
Comment (singiamtel) [inline PWGDQ/TableProducer/tableMakerMC_withAssoc.cxx:168]:
```suggestion
  Configurable<bool> fRefitGlobalMuon{"cfgRefitGlobalMuon", true, "Propagate muon tracks through absorber (do not use if applying pairing)"};
```

**Review by singiamtel — COMMENTED** [review]
Small typo on the copy paste

**Comment by iarsene** [comment]
Hi, its fine for me. But since it deals with muon data, I hope Luca can have a look and approve. If he is away, just ping me and I will approve.

**Comment by lucamicheletti93** [comment]
@mcoquet642 thanks for the fix, I approve the PR!

**Review by lucamicheletti93 — APPROVED** [review]

**Pull request merged**